tokio-util/src/sync/mpsc.rs

```diff
@@ -64,13 +64,45 @@ async fn make_acquire_future<T>(
     }
 }
 
-impl<T: Send + 'static> PollSender<T> {
+#[derive(Debug)]
+// TODO: This should be replace with a type_alias_impl_trait to eliminate `'static` and all the transmutes
+struct PollSenderFuture<T>(ReusableBoxFuture<'static, Result<OwnedPermit<T>, PollSendError<T>>>);
```

Comment:
You may be able to do this simpler by adding an `unsafe_set` method to `ReusableBoxFuture` that accepts any future, even if it isn't `Send` or `'static`.

Reply:
I can do this, but it didn't seem to simplify things. Let me know if you disagree.

Reply:
It's up to you. I suggested it because you could drop the `PollSenderFuture` type.

Reply:
I kind of like the `PollSenderFuture` as I think it's needed in the future `type_alias_impl_trait` implementation.